dingo/gw/noise/utils.py

```diff
@@ -0,0 +1,245 @@
+
+    for detector in settings["detectors"]:
+
+        key = run + "_" + detector
+        urls = URL_DIRECTORY[key]
+
+        starts, stops, durations = [], [], []
+        for url in urls:
+            r = requests.get(url, allow_redirects=True)
+            c = StringIO(r.content.decode("utf-8"))
+            starts_seg, stops_seg, durations_seg = np.loadtxt(
+                c, dtype="int", unpack=True
+            )
+            starts = np.hstack([starts, starts_seg])
+            stops = np.hstack([stops, stops_seg])
+            durations = np.hstack([durations, durations_seg])
+
+        segs = np.array(list(zip(starts, stops, durations)), dtype=int)
+        segs = segs[segs[:, 2] >= time_psd]
+
+        valid_segments = []
+        for idx in range(segs.shape[0]):
+            seg = segs[idx, :]
+            start_time = seg[0]
+            end_time = start_time + time_psd
+
+            while end_time <= seg[1]:
+                # check that no event is present in the segments
+                if not any(start_time <= event <= end_time for event in event_list):
+                    valid_segments.append((start_time, end_time))
+                start_time = end_time + time_gap
+                end_time = start_time + time_psd
+
+        if num_psds_max is not None and num_psds_max > 0:
+            valid_segments = valid_segments[:num_psds_max]
+
+        time_segments[detector] = valid_segments
+
+    return time_segments
+
+
+def merge_datasets(asd_dataset_list):
+    """
+    Merges a list of asd datasets into ont
+    Parameters
+    ----------
+    asd_dataset_list: List of ASDDatasets to be merged
+
+    Returns
+    -------
+    A single combined ASDDataset object
+    """
+
+    merged_dict = {"asds": {}, "gps_times": {}}
+
+    for det, asd_list in asd_dataset_list.items():
+        print(f"Merging {len(asd_list)} datasets into one for detector {det}.")
+        merged_dict["asds"][det] = np.vstack(
+            [asd_dataset.asds[det] for asd_dataset in asd_list]
+        )
+        merged_dict["gps_times"][det] = np.hstack(
+            [asd_dataset.gps_times[det] for asd_dataset in asd_list]
+        )
+
+    # copy settings from last dataset
+    merged_dict["settings"] = copy.deepcopy(asd_list[-1].settings)
+
+    merged = ASDDataset(dictionary=merged_dict)
+    return merged
+
+
+def merge_datasets_cli():
+    """
+    Command-line function to combine a collection of datasets into one. Used for
+    parallelized ASD dataset generation.
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--data_dir",
+        type=str,
+        required=True,
+        help="Path where the PSD data is to be stored. Must contain a 'settings.yaml' file.",
+    )
+    parser.add_argument(
+        "--settings_file",
+        type=str,
+        required=True,
+        help="Path to a settings file in case two different datasets are generated in the same directory",
+    )
```

Comment:
Same issue as above: the settings file is not optional, so why should data-dir contain a settings file?